File: Data Structures & Algorithms/find-duplicate-integer/submission-0.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    findDuplicate(nums) {
        const mapper = new Map();

        for (let i = 0; i < nums.length; i++) {
            if (mapper.get(nums[i]) === undefined) {
                mapper.set(nums[i], 1);
            } else {
                return nums[i];
            }
        }
    }
}

File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        # Create a dictionary to collect frequency as value and number as a key
        dic = {}
        
        # Create a list to collect the result
        result = []

        # iterate each number and count the frequency
        for n in nums:
            if n in dic:
                dic[n] += 1
            else:
                dic[n] = 1
        
        # Create a while loop until k becomes none
        while k != 0:
            # get most frequent number as a key
            n = max(dic, key=dic.get)

            # make the value as 0 zero
            dic[n] = 0
            
            # append the number to list
            result.append(n)

            # decrease k by one
            k -= 1
        
        return result


        